tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Tests/CoreTests/PlanConfigurationTests.cs

```diff
@@ -5,29 +5,42 @@ namespace Azure.Sdk.Tools.SecretRotation.Tests.CoreTests;
 public class PlanConfigurationTests
 {
     [Test]
-    public void FromFile_MissingFile_ThrowsException()
+    public void TryLoadFromFile_MissingFile_ThrowsException()
     {
         string configurationPath = TestFiles.ResolvePath("TestConfigurations/missing.json");
 
-        Assert.Throws<RotationConfigurationException>(() => PlanConfiguration.FromFile(configurationPath));
+        Assert.Throws<RotationConfigurationException>(() => PlanConfiguration.TryLoadFromFile(configurationPath, out _));
     }
 
     [Test]
-    public void FromFile_InvalidPath_ThrowsException()
+    public void TryLoadFromFile_InvalidPath_ThrowsException()
     {
         string configurationPath = @"&invalid:path?";
 
-        Assert.Throws<RotationConfigurationException>(() => PlanConfiguration.FromFile(configurationPath));
+        Assert.Throws<RotationConfigurationException>(() => PlanConfiguration.TryLoadFromFile(configurationPath, out _));
     }
 
     [Test]
-    public void FromFile_ValidPath_ReturnConfiguration()
+    public void TryLoadFromFile_WrongSchema_ReturnsFalse()
+    {
+        string configurationPath = TestFiles.ResolvePath("TestConfigurations/Invalid/wrong-schema.json");
+
+        // Act
+        bool success = PlanConfiguration.TryLoadFromFile(configurationPath, out var configuration);
+
+        Assert.False(success);
+        Assert.Null(configuration);
+    }
+    
+    [Test]
+    public void TryLoadFromFile_ValidPath_ReturnConfiguration()
     {
         string configurationPath = TestFiles.ResolvePath("TestConfigurations/Valid/random-string.json");
 
         // Act
-        PlanConfiguration configuration = PlanConfiguration.FromFile(configurationPath);
+        bool success = PlanConfiguration.TryLoadFromFile(configurationPath, out var configuration);
 
+        Assert.True(success);
```

Comment:
Is there any value in adding tag tests as well?

Reply:
There is, but I'd want to mock the file system.  I wouldn't want the name/tag filtering test to be affected by test files being added or removed from the folder.

Reply:
If you fed an array of de-serialized objects to the tag filter method, and checked the output, that wouldn't have to touch the fs.

Reply:
I went ahead with adding another test config folder named "TagMatching". any changes in the folder could cause issues with tag related tests.  We'll catch them in PR if it happens

Reply:
Tag and name filtering happens as 2 linq statements in the large config reader method:

```cs
        if (names.Any())
        {
            planConfigurations = planConfigurations
                .Where(plan => names.Contains(plan.Name, StringComparer.OrdinalIgnoreCase));
        }

        if (tags.Any())
        {
            planConfigurations = planConfigurations
                .Where(plan => tags.All(tag => plan.Tags.Contains(tag, StringComparer.OrdinalIgnoreCase)));
        }
```

I could extract those lines to a separate testable method, but at that point, we'd mostly be testing LINQ.